tests/test_0action_mask.py

```diff
@@ -0,0 +1,259 @@
+        obs, _, done, infos = env.step(action)
+
+        action_masks.clear()
+        for info in infos:
+            env_action_mask = info.get('action_mask')
+            action_masks.append(env_action_mask)
+
+    env.close()
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize('vec_env', [SubprocVecEnv, DummyVecEnv])
+@pytest.mark.parametrize('policy', [MlpPolicy])
+@pytest.mark.parametrize('env_class', [DiscreteActionMaskEnv, MultiDiscreteActionMaskEnv,
+                                       MultiDiscreteUnbalancedActionMaskEnv])
+def test_action_mask_learn_ppo1(vec_env, policy, env_class):
+    env = vec_env([env_class])
+
+    model = PPO1(policy, env, verbose=0)
+    model.learn(total_timesteps=500)
+    env.close()
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize('vec_env', [SubprocVecEnv, DummyVecEnv])
+@pytest.mark.parametrize('policy', [MlpPolicy])
+@pytest.mark.parametrize('env_class', [DiscreteActionMaskEnv, MultiDiscreteActionMaskEnv,
+                                       MultiDiscreteUnbalancedActionMaskEnv])
+def test_action_mask_run_ppo1(vec_env, policy, env_class):
+    env = vec_env([env_class])
+
+    model = PPO1(policy, env, verbose=0)
+
+    obs, done, action_masks = env.reset(), [False], []
+    while not done[0]:
+        action, _states = model.predict(obs, action_mask=action_masks)
+        obs, _, done, infos = env.step(action)
+
+        action_masks.clear()
+        for info in infos:
+            env_action_mask = info.get('action_mask')
+            action_masks.append(env_action_mask)
+
+    env.close()
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize('vec_env', [SubprocVecEnv, DummyVecEnv])
+@pytest.mark.parametrize('policy', [MlpPolicy])
+@pytest.mark.parametrize('env_class', [DiscreteActionMaskEnv, MultiDiscreteActionMaskEnv,
+                                       MultiDiscreteUnbalancedActionMaskEnv])
+def test_action_mask_learn_ppo2(vec_env, policy, env_class):
+    env = vec_env([env_class] * 2)
+
+    model = PPO2(policy, env, verbose=0, nminibatches=2)
+    model.learn(total_timesteps=1000)
+    env.close()
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize('vec_env', [SubprocVecEnv, DummyVecEnv])
+@pytest.mark.parametrize('policy', [MlpPolicy])
+@pytest.mark.parametrize('env_class', [DiscreteActionMaskEnv, MultiDiscreteActionMaskEnv,
+                                       MultiDiscreteUnbalancedActionMaskEnv])
+def test_action_mask_run_ppo2(vec_env, policy, env_class):
+    env = vec_env([env_class])
+
+    model = PPO2(policy, env, verbose=0, nminibatches=1)
+
+    obs, done, action_masks = env.reset(), [False], []
+    while not done[0]:
+        action, _states = model.predict(obs, action_mask=action_masks)
+        obs, _, done, infos = env.step(action)
+
+        action_masks.clear()
+        for info in infos:
+            env_action_mask = info.get('action_mask')
+            action_masks.append(env_action_mask)
+
+    env.close()
+
```

Comment:
@quoctuams, 
Do you input the action masks to predict function?

Reply:
My bad, I realized that just now =_= I will add that and report back the result. Thanks for you guys' contribution :)